### PR TITLE
#195 | Implement Custom Validation for First and Last Name Fields

### PR DESCRIPTION
**Pull request by AhmetAksunger**
# PULL REQUEST TEMPLATE

## Description

Added a custom validation for the request classes that contain first & last name fields

## Related Issue

#195 

## Proposed Changes

Please provide a brief and descriptive title for your pull request, summarizing the changes or features introduced. It
should be in English and clearly indicate the purpose of the pull request. Optionally, you can provide a more detailed
description of what you have done.

- Branches should follow the below `Naming Conventions of Branchs` header in [README.md](https://github.com/afet-yonetim-sistemi/ays-be/edit/main/README.md) File

## Additional Information

If there's any additional information or context that the reviewers should know, please provide it here.

## Checklist

Please check the following before submitting your pull request:

- [x] I have tested my changes locally and they are working as expected.
- [x] The code is formatted according to the project's coding guidelines and style.
- [x] Necessary documentation has been added or existing documentation has been updated.
- [x] Relevant unit tests have been written and included.
- [x] Relevant integration tests have been written and included.
- [x] The code has been reviewed to ensure its quality.
- [x] The code does not contain any issues flagged by SonarLint.
- [x] Default reviewers have been assigned to this pull request.
- [x] Assignees have been added if necessary.
- [x] Labels have been applied if necessary.
- [x] If an issue is opened and this pull request is related to that issue, it should not be associated with any project
  board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project
  board (BE Board, In Review column) and milestone to track the progress.

## Reviewers

The default project maintainers and team members will be automatically assigned as reviewers for this pull request.

### CODE REVIEW

File: src/main/java/com/ays/common/util/validation/NameValidator.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     *
+     * @param value   object to validate
+     * @param context context in which the constraint is evaluated
+     * @return true if the value is valid, false otherwise
+     */
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        return value.matches(NAME_REGEX);
+    }
+}
```
Comment (agitrubard) [inline src/main/java/com/ays/common/util/validation/NameValidator.java:52]: Örneklerle beraber güzel olmuş eline sağlık 🙌🏼 

Ek olarak isValid() metodunda bir tane null check ekleyebiliriz, çünkü null olan değerler buradaki validasyona takılmamalı, tam tersi `@NotNull` validasyonuna takılmalı. 

Örnek olarak aşağıdaki gibi;
```
if (!org.springframework.util.StringUtils.hasText(value)) {
   return true;
}
```

File: src/main/java/com/ays/common/util/validation/NameValidator.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+/**
+ * A custom validator implementation for the {@link Name} annotation.
+ * Validates whether the provided first/last name is a valid name based on the
+ * specified regular expression.
+ */
+public class NameValidator implements ConstraintValidator<Name, String> {
```
Comment (agitrubard) [inline src/main/java/com/ays/common/util/validation/NameValidator.java:11]: Validator sınıfları `package-private` yapabiliriz çünkü dışarıdan çağrılmaya ihtiyaç olmayacak

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/main/java/com/ays/common/util/validation/NameValidator.java (hunk 1 above)
Comment (AhmetAksunger) [reply src/main/java/com/ays/common/util/validation/NameValidator.java:56]: Teşekkürler, düzelttim. @agitrubard

File: src/main/java/com/ays/common/util/validation/NameValidator.java (hunk 2 above)
Comment (AhmetAksunger) [reply src/main/java/com/ays/common/util/validation/NameValidator.java]: Düzelttim. @agitrubard

**Review by agitrubard — DISMISSED** [review]

**Review by agitrubard — APPROVED** [review]

**Review by moaydogdu — APPROVED** [review]

**Review by anilhazar — APPROVED** [review]

**Pull request merged**